src/placeos-rest-api/controllers/repositories.cr

```diff
@@ -103,7 +103,10 @@ module PlaceOS::Api
 
     def self.pull_repository(repository : Model::Repository)
       # Trigger a pull event
-      spawn { repository.pull! }
+      spawn do
+        sleep 0.1
```

Comment:
Why did you put a sleep here?

Reply:
as the `await_model_change` method spawns and then makes blocking database request in that spawn.
So this spawn is scheduled before the await_model_change spawn and then a database request is made which causes another fiber schedule. So we need to give it some time to setup.

Reply:
100 milliseconds feels excessive. I feel like using a sleep here is a smell. Can you achieve this through Fiber.yield/Fiber.enqueue

Reply:
not really as the database request will take a few ms
a pull will take longer than 100ms anyway so this only really has any impact on the no change path and users are expecting to wait for a pull.

We can do better with a refactor of the `await_model_change` method, which would need another block added.
* monitor
* check callback (is this the change we were waiting for)
* action callback (the pull method)

Reply:
actually even then I think it's hard to know once the database monitoring has been established
I think 100ms is fine

Reply:
I can't think of a better option than sleep to handle the database request.
We could probably lower it to 50ms